src/LeanCode.Contracts.Admin/AdminQuery.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace LeanCode.Contracts.Admin;
+
+public abstract class AdminQuery<TResult> : IQuery<AdminQueryResult<TResult>>
+{
+    public int Page { get; set; }
+    public int PageSize { get; set; }
+
+    public SortOrderDTO? SortOrder { get; set; }
+    public string? SortBy { get; set; }
+}
+
+public enum SortOrderDTO
+{
+    Descending = 0,
+    Ascending = 1
+}
+
+public class AdminQueryResult<TResult>
+{
+    public long Total { get; set; }
+    public List<TResult> Items { get; set; }
+}
+
+public class AdminFilterRange<T>
+{
+    public T? From { get; set; }
+    public T? To { get; set; }

```

Comment:
Are those inclusive or exclusive? Might be a good idea to state this explicitly, either via better property names or just `<remarks>`.

Reply:
It's not specified. Values of `From` and `To` are values that user inputs and how they are interpreted depends on the person writing query handler.